libcodechecker/cmd/product_client.py

```diff
@@ -33,16 +35,29 @@ def handle_list_products(args):
             results.append({product.endpoint: product})
         print(CmdLineOutputEncoder().encode(results))
     else:  # plaintext, csv
-        header = ['Status', 'Endpoint', 'Name', 'Description']
+        header = ['Database status', 'Endpoint', 'Name', 'Description']
         rows = []
         for product in products:
             name = base64.b64decode(product.displayedName_b64) \
                 if product.displayedName_b64 else ''
             description = base64.b64decode(product.description_b64) \
                 if product.description_b64 else ''
-            rows.append(('Database error' if not product.connected
-                         else 'No access' if not product.accessible
-                         else '',
+
+            db_status_msg = 'OK'
+            db_status = product.databaseStatus
+            if db_status == DBStatus.MISSING:
+                db_status_msg = 'Missing database'
+            elif db_status == DBStatus.FAILED_TO_CONNECT:
+                db_status_msg = 'Connection failed'
+            elif db_status == DBStatus.SCHEMA_MISMATCH:
+                db_status_msg = 'Database schema mismatch'
+            elif db_status == DBStatus.SCHEMA_INITIALIZATION_ERROR:
+                db_status_msg = 'Schema initialization error'
+
+            if db_status == DBStatus.OK and not product.accessible:
```

Comment:
The fact that the user has no access to the database is irrelevant to the database's status itself, as this value is stored in the config db. This condition hides the "even if the database would reconnect, you'd still have no access..." state.